enterprise/frontend/src/metabase-enterprise/data-studio/common/components/UnpublishTablesModal/UnpublishTablesModal.tsx

```diff
@@ -0,0 +1,103 @@
+import { msgid, ngettext, t } from "ttag";
+
+import {
+  Form,
+  FormErrorMessage,
+  FormProvider,
+  FormSubmitButton,
+} from "metabase/forms";
+import { useMetadataToasts } from "metabase/metadata/hooks";
+import { Box, Button, Group, Modal, Stack, Text, rem } from "metabase/ui";
+import { useUnpublishTablesMutation } from "metabase-enterprise/api";
+import type { DatabaseId, SchemaId, TableId } from "metabase-types/api";
+
+type UnpublishTablesModalProps = {
+  databaseIds?: DatabaseId[];
+  schemaIds?: SchemaId[];
+  tableIds?: TableId[];
+  onUnpublish?: () => void;
+  onClose: () => void;
+};
+
+export function UnpublishTablesModal({
+  databaseIds = [],
+  schemaIds = [],
+  tableIds = [],
+  onUnpublish,
+  onClose,
+}: UnpublishTablesModalProps) {
+  const [unpublishTables] = useUnpublishTablesMutation();
+  const { sendSuccessToast } = useMetadataToasts();
+
+  const handleSubmit = async () => {
+    const action = unpublishTables({
+      database_ids: databaseIds,
+      schema_ids: schemaIds,
+      table_ids: tableIds,
+    });
+    await action.unwrap();
+    sendSuccessToast(t`Unpublished`);
+    onUnpublish?.();
+    onClose();
+  };
+
+  return (
+    <Modal
+      title={getTitle(databaseIds, schemaIds, tableIds)}
+      opened
```

Comment:
Super minor, but can we start leaving modals mounted and actually passing opened in? Rather than mounting them with opened hardcoded?

Reply:
Fixed